tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ServicesCommand.java

```diff
@@ -0,0 +1,286 @@
+        if( supervisor.getOutInfo().equals( "No exported packages\n" ) )
+        {
+            supervisor.getAgent().stdin(
+                "services " + "(objectClass=" + serviceName + ")" + " | grep \"Registered by bundle:\" " );
+            serviceBundleInfo = new ServiceParser().parseRegisteredBundle( supervisor.getOutInfo() );
+        }
+        else
+        {
+            serviceBundleInfo = new ServiceParser().parseSymbolicName( supervisor.getOutInfo() );
+        }
+
+        return serviceBundleInfo;
+    }
+
+    public class ServicesSupervisor extends AgentSupervisor<Supervisor, Agent> implements Supervisor
+    {
+
+        private String outinfo;
+
+        @Override
+        public boolean stdout( String out ) throws Exception
+        {
+            if( !"".equals( out ) && out != null )
+            {
+                out = out.replaceAll( ".*>.*$", "" );
+                if( !out.equals( "" ) && !out.equals( "true\n" ) && !out.equals( "false\n" ) )
+                {
+                    outinfo = out;
+                }
+            }
+            return true;
+        }
+
+        @Override
+        public boolean stderr( String out ) throws Exception
+        {
+            return true;
+        }
+
+        public void connect( String host, int port ) throws Exception
+        {
+            super.connect( Agent.class, this, host, port );
+        }
+
+        @Override
+        public void event( Event e ) throws Exception
+        {
+        }
+
+        public String getOutInfo()
+        {
+            return outinfo;
+        }
+
+    }
+
+    private class ServiceParser
+    {
+
+        public String[] parseService( String outinfo )
+        {
+
+            final Pattern pattern = Pattern.compile( "(?<=\\{)(.+?)(?=\\})" );
+            final Matcher matcher = pattern.matcher( outinfo );
+            final List<String> ls = new ArrayList<>();
+            while( matcher.find() )
+            {
+                ls.add( matcher.group() );
+            }
+
+            Iterator<String> iterator = ls.iterator();
+            while( iterator.hasNext() )
+            {
+                iterator.next();
+                if( iterator.hasNext() )
+                {
+                    iterator.next();
+                    iterator.remove();
+                }
+            }
+            final List<String> listservice = new ArrayList<>();
+            for( String bs : ls )
+            {
+                if( bs.split( "," ).length > 1 )
+                {
+                    for( String bbs : bs.split( "," ) )
+                    {
+                        listservice.add( bbs.trim() );
+                    }
+                }
+                else
+                {
+                    listservice.add( bs );
+                }
+            }
+
+            final Set<String> set = new HashSet<String>();
+            final List<String> newList = new ArrayList<>();
```

Comment:
Always put a newline before a "for" loop unless the for loop is the first statement in the block
```
if(...)
{
   for(...)
```
and

```
foo();
bar();

for(...)
{
```